migration/v0.30/apply/90-psp-violations.sh

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+
+ROOT="$(readlink -f "$(dirname "${0}")/../../../")"
+
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+declare -a results
+
+function set_violating_resources() {
+  results=()
+  cluster="$1"
+
+  # Get violations for PSPs
+  violations=$(kubectl_do "$cluster" get constraints -o yaml | yq4 '[.items[] | select(.kind == "K8sPSP*") | .status.violations[]]')
+
+  # Build array of maps with relevant information
+  resources=$(echo "$violations" | yq4 '.[] |[{"name": .name, "namespace": .namespace}]' | yq4 'unique_by(.name)')
+
+  # Create bash array to be able to loop over each entry
+  readarray resource_arr < <(echo "$resources" | yq4 e -o=j -I=0 '.[]')
+
+  for resource in "${resource_arr[@]}"; do
+    namespace=$(echo "$resource" | yq4 e '.namespace')
+    pod_name=$(echo "$resource" | yq4 e '.name')
+
+    owner_reference=$(kubectl_do "$cluster" -n "$namespace" get pod "$pod_name" -oyaml | yq4 '.metadata.ownerReferences.[0]')
+
+    # Skip standalone pods
+    if [ -z "$owner_reference" ]; then continue; fi
+
+    owner_kind=$(echo "$owner_reference" | yq4 .kind)
+    owner_name=$(echo "$owner_reference" | yq4 .name)
+
+    # Get owner of ReplicaSets
+    if [ "$owner_kind" == "ReplicaSet" ]; then
+      owner_reference=$(kubectl_do "$cluster" -n "$namespace" get rs "$owner_name" -oyaml | yq4 '.metadata.ownerReferences.[0]')
+
+      # Skip standalone ReplicaSets
+      if [ -z "$owner_reference" ]; then continue; fi
+
+      owner_kind=$(echo "$owner_reference" | yq4 .kind)
+      owner_name=$(echo "$owner_reference" | yq4 .name)
+    fi
+
+    results+=('{name: '"$owner_name"', namespace: '"$namespace"', kind: '"$owner_kind"'}')
+  done
+
+  # Convert array to multiline string and keep only unique lines
+  # shellcheck disable=SC2178
+  results=$(printf "%s\n" "${results[@]}")
+  # shellcheck disable=SC2178
+  # shellcheck disable=SC2128
+  results=$(echo "$results" | sort -u)
+}
+
+function restart_violating_resources() {
+  IFS=$'\n';
+  # shellcheck disable=SC2128
+  for entry in $results; do
```

Comment:
Can you not try to fix the spellchecks?
```suggestion
  for entry in "${results}"; do
```

Reply:
Need word splitting here

Reply:
Got rid of the other ones